pkg/cninetwork/cni_network.go

```diff
@@ -165,6 +165,14 @@ func GetIPfromPID(pid int) (*net.IP, error) {
 
 }
 
+// CNIGateway returns the gateway for default subnet
+func CNIGateway() string {
+	ip, _, _ := net.ParseCIDR(defaultSubnet)
+	ip = ip.To4()
+	ip[3] = 1
```

Comment:
Please use range checking. Return an error, don't swallow it.

Reply:
Fixed.

Reply:
I can't see any range-checking here at all. We may not have an index of [3]